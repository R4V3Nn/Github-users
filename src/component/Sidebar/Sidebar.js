import React, { Component } from 'react';
import { browserHistory } from 'react-router';
import { SidePanel } from '@talend/react-components';

const actions = [
	{
		label: 'About',
		icon: 'talend-user-circle',
		onClick: () => browserHistory.push('/about'),
	},
	{
		label: 'Repos',
		icon: 'talend-hierarchical-view',
		onClick: () => browserHistory.push('/repos'),
	},
	{
		label: 'Users',
		icon: 'talend-group-circle',
		onClick: () => browserHistory.push('/users'),
	},
];
export default class Sidebar extends Component {
	constructor() {
		super();
		this.state = {
			docked: false,
		};
	}
	toggleDock() {
		this.setState({
			docked: !this.state.docked,
		});
	}

	render() {
		return (
			<SidePanel
				actions={actions}
				docked={this.state.docked}
				onToggleDock={() => this.toggleDock()}
			/>
		);
	}
}

