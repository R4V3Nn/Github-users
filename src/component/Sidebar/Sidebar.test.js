import React from 'react';
import { shallow } from 'enzyme';

import Sidebar from './Sidebar';

describe('Sidebar component', () => {
	const props = {
		actions: [
			{
				label: 'About',
				icon: 'talend-user-circle',
				onClick: () => jest.fn(),
			},
			{
				label: 'Repos',
				icon: 'talend-hierarchical-view',
				onClick: () => jest.fn(),
			},
		],
		docked: false,
		onToggleDock: jest.fn(),
	};


	it('should render Sidebar component', () => {
		// when
		const wrapper = shallow(<Sidebar {...props} />);

		// then
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
