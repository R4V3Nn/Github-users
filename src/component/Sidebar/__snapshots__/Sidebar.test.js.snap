// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component should render Sidebar component 1`] = `
<Translate(SidePanel)
  actions={
    Array [
      Object {
        "icon": "talend-user-circle",
        "label": "About",
        "onClick": [Function],
      },
      Object {
        "icon": "talend-hierarchical-view",
        "label": "Repos",
        "onClick": [Function],
      },
      Object {
        "icon": "talend-group-circle",
        "label": "Users",
        "onClick": [Function],
      },
    ]
  }
  docked={false}
  onToggleDock={[Function]}
/>
`;
